nixos/modules/services/misc/matrix-synapse.nix

```diff
@@ -5,17 +5,22 @@ with lib;
 let
   cfg = config.services.matrix-synapse;
   logConfigFile = pkgs.writeText "log_config.yaml" cfg.logConfig;
+  mkResource = r: ''{names: [${concatStringsSep "," r.names}], compress: ${if r.compress then "true" else "false"}}'';
+  mkListener = l: ''{port: ${l.port}, bind_address: ${l.bind_address}, type: ${l.type}, tls: ${if l.tls then "true" else "false"}, x_forwarded: ${if l.x_forwarded then "true" else "false"}, resources: [${concatStringsSep "," (map mkResource l.resources)}]}'';
```

Comment:
if bind_address is empty, it also crashes matrix_synapse because `bind_address: ,` is translated to NoneType instead of an empty string (derp). It'd be a good idea to surround ${l.bind_address} with quotes.

Alternatively, why not use builtins.toJSON ? That should make everything work correctly and would be arguably "cleaner".